practice/blackjack/deck.py

```diff
@@ -0,0 +1,135 @@
+"""Module containing the Deck class"""
+
+
+from card import Card
+from random import sample
+
+
+class Deck:
+    """A class for representing a Deck of cards"""
+
+    def __init__(self, cards):
+        self.cards = cards
+
+    def __eq__(self, other):
+        return (
+            self.cards == other.cards
+            )
+
+    def __repr__(self):
+        return 'Deck({!r})'.format(
+            self.cards
+        )
+
+
+STANDARD_DECK = Deck([
+    Card('A', 'C'),
+    Card('2', 'C'),
+    Card('3', 'C'),
+    Card('4', 'C'),
+    Card('5', 'C'),
+    Card('6', 'C'),
+    Card('7', 'C'),
+    Card('8', 'C'),
+    Card('9', 'C'),
+    Card('10', 'C'),
+    Card('J', 'C'),
+    Card('Q', 'C'),
+    Card('K', 'C'),
+    Card('A', 'D'),
+    Card('2', 'D'),
+    Card('3', 'D'),
+    Card('4', 'D'),
+    Card('5', 'D'),
+    Card('6', 'D'),
+    Card('7', 'D'),
+    Card('8', 'D'),
+    Card('9', 'D'),
+    Card('10', 'D'),
+    Card('J', 'D'),
+    Card('Q', 'D'),
+    Card('K', 'D'),
+    Card('A', 'H'),
+    Card('2', 'H'),
+    Card('3', 'H'),
+    Card('4', 'H'),
+    Card('5', 'H'),
+    Card('6', 'H'),
+    Card('7', 'H'),
+    Card('8', 'H'),
+    Card('9', 'H'),
+    Card('10', 'H'),
+    Card('J', 'H'),
+    Card('Q', 'H'),
+    Card('K', 'H'),
+    Card('A', 'S'),
+    Card('2', 'S'),
+    Card('3', 'S'),
+    Card('4', 'S'),
+    Card('5', 'S'),
+    Card('6', 'S'),
+    Card('7', 'S'),
+    Card('8', 'S'),
+    Card('9', 'S'),
+    Card('10', 'S'),
+    Card('J', 'S'),
+    Card('Q', 'S'),
+    Card('K', 'S')
+    ])
+EXAMPLE_DECK = Deck([Card('A', 'C'), Card('2', 'C'), Card('3', 'C')])
+
+
+def get_shuffled_deck(deck):
+    """Shuffles a deck of cards
+
+    >>> get_shuffled_deck(EXAMPLE_DECK) == EXAMPLE_DECK
+    False
+    """
+    shuffled_cards = [sample(deck.cards, 1)]
+    shuffled_deck = Deck(shuffled_cards)
+    return shuffled_deck
```

Comment:
I don't think this function does what you want it to.

```py
>>> from random import sample
>>> sample([1, 2, 3, 4], 1)
[3]
```

`sample()` returns a list of length of the second argument. So `shuffled_cards` is a nested list of a single item in the inner list. And you're putting that in the resulting deck.

I think you want to use `random.shuffle()`.

Also, this doctest is **flakey** which means there's a chance it will fail, but not because the code is bad. Even if you were shuffling correctly, there is a chance the order will be unchanged and the test will randomly fail. I would just not test a function that deals with randomness and instead isolate it as much as possible.